Stripe/STPSource.m

```diff
@@ -289,15 +287,38 @@ - (UIImage *)templateImage {
 }
 
 - (NSString *)label {
-    if (self.type == STPSourceTypeCard
-        && self.cardDetails != nil) {
-        NSString *brand = [STPCard stringFromBrand:self.cardDetails.brand];
-        return [NSString stringWithFormat:@"%@ %@", brand, self.cardDetails.last4];;
-    }
-    else {
-        return [STPCard stringFromBrand:STPCardBrandUnknown];
+    switch (self.type) {
+        case STPSourceTypeBancontact:
+            return @"Bancontact";
+        case STPSourceTypeCard:
+            if (self.cardDetails != nil) {
+                NSString *brand = [STPCard stringFromBrand:self.cardDetails.brand];
+                return [NSString stringWithFormat:@"%@ %@", brand, self.cardDetails.last4];
+            }
+            else {
+                return [STPCard stringFromBrand:STPCardBrandUnknown];
+            }
+        case STPSourceTypeGiropay:
+            return @"Giropay";
+        case STPSourceTypeIDEAL:
+            return @"iDEAL";
+        case STPSourceTypeSEPADebit:
+            return @"SEPA Direct Debit";
+        case STPSourceTypeSofort:
+            return @"SOFORT";
+        case STPSourceTypeThreeDSecure:
+            return @"3D Secure";
+        case STPSourceTypeAlipay:
+            return @"Alipay";
+        case STPSourceTypeP24:
+            return @"P24";
+        case STPSourceTypeEPS:
+            return @"EPS";
+        case STPSourceTypeMultibanco:
+            return @"Multibanco";
+        case STPSourceTypeUnknown:
+            return [STPCard stringFromBrand:STPCardBrandUnknown];
```

Comment:
wdyt about using `@"Unknown"` here instead of this layer of indirection with `STPCardBrandUnknown`?

Reply:
Yeah why not. I first put this here because I was focused on card source types but it is a little out of place in the `STPSourceTypeUnknown` block.